src/components/CIPipelineN/CIPipeline.tsx

```diff
@@ -0,0 +1,767 @@
+                        delete variableDetail.refVariableStage
+                    }
+                }
+            }
+        }
+        if (isFromAddNewTask) {
+            _inputVariablesListPerTask.push(new Map(_outputVariablesFromPrevSteps))
+        }
+        const _inputVariablesListFromPrevStep = { ...inputVariablesListFromPrevStep }
+        _inputVariablesListFromPrevStep[activeStageName] = _inputVariablesListPerTask
+        setInputVariablesListFromPrevStep(_inputVariablesListFromPrevStep)
+        setFormDataErrorObj(_formDataErrorObj)
+        return { index: i + 1, calculatedStageVariables: _inputVariablesListPerTask }
+    }
+
+    const addNewTask = () => {
+        const _formData = { ...formData }
+        const detailsFromLastStep = calculateLastStepDetail(true, _formData, activeStageName)
+        const stage = {
+            id: detailsFromLastStep.index,
+            index: detailsFromLastStep.index,
+            name: `Task ${detailsFromLastStep.index}`,
+            description: '',
+            stepType: '',
+            directoryPath: '',
+        }
+        _formData[activeStageName].steps.push(stage)
+        setFormData(_formData)
+        const _formDataErrorObj = { ...formDataErrorObj }
+        _formDataErrorObj[activeStageName].steps.push({
+            name: { isValid: true, message: null },
+            isValid: true,
+        })
+        setFormDataErrorObj(_formDataErrorObj)
+        setSelectedTaskIndex(_formData[activeStageName].steps.length - 1)
+    }
+
+    const getNavLink = (toLink: string, stageName: string) => {
+        return (
+            <li className="tab-list__tab">
+                <NavLink
+                    replace
+                    className="tab-list__tab-link fs-13 pt-5 pb-5 flexbox"
+                    activeClassName="active"
+                    to={toLink}
+                    onClick={() => {
+                        validateStage(activeStageName, formData)
+                    }}
+                >
+                    {BuildTabText[stageName]}
+                    {!formDataErrorObj[stageName].isValid && <AlertTriangle className="icon-dim-16 mr-5 ml-5 mt-3" />}
+                </NavLink>
+            </li>
+        )
+    }
+
+    return (
+        <VisibleModal className="">
+            {' '}
+            <div
+                className={`modal__body modal__body__ci_new_ui br-0 modal__body--p-0 ${
+                    isAdvanced ? 'advanced-option-container' : 'bottom-border-radius'
+                }`}
+            >
+                <div className="flex flex-align-center flex-justify bcn-0 pr-20">
+                    <h2 className="fs-16 fw-6 lh-1-43 m-0 title-padding">{title}</h2>
+                    <button
+                        type="button"
+                        className="transparent flex icon-dim-24"
+                        onClick={() => {
+                            close()
+                        }}
```

Comment:
`close()` Can be used directly instead of wrapping inside another? i.e. `onClick={close}`

Reply:
If we use directly we will be getting event as params. which we don't need, we have to set a boolean in some cases of close.